src/main/cpp/src/TaskPriorityJni.cpp

```diff
@@ -20,14 +20,39 @@
 
 namespace spark_rapids_jni {
 
+// Track the next priority to assign and maintain a map of attempt_id to priority
+static long next_task_priority = std::numeric_limits<long>::max() - 1;
+static std::mutex priority_mutex;
+static std::unordered_map<long, long> attempt_priorities;
```

Comment:
These static variables are directly under the `spark_rapids_jni` namespace, which are visible and can be mixed with all other JNI functions. This is not a good encapsulation practice. How about put them into some other namespace, like this?
```
namespace task_priority {
static long ....;
}
```

Reply:
Another option is that, we put them into an anonymous namespace, which guarantees them to be visible **only** to the function in this source file. I think this is a better option because they should not be visible to any other functions in other places (shouldn't they?).